runners/flink/src/main/java/org/apache/beam/runners/flink/translation/functions/FlinkMergingNonShuffleReduceFunction.java

```diff
@@ -83,7 +83,7 @@ public void reduce(
         new FlinkSideInputReader(sideInputs, getRuntimeContext());
 
     AbstractFlinkCombineRunner<K, InputT, AccumT, OutputT, W> reduceRunner;
-    if (windowingStrategy.getWindowFn().windowCoder().equals(IntervalWindow.getCoder())) {
+    if (windowingStrategy.getWindowFn() instanceof Sessions) {
```

Comment:
I think you want to keep `needsMerge` also. According to the dev list discussion, I changed it so that if the `windowFn` is sessions and `needsMerge` is false, you do the non-sorting (non-merging) group by key and window.

Reply:
Done.